tests/test_arrow.py

```diff
@@ -1197,6 +1253,30 @@ def test_bounds_param_is_passed(self):
             (arrow.Arrow(2013, 5, 5, 16), arrow.Arrow(2013, 5, 5, 18)),
         ]
 
+    def test_exact(self):
+        result = list(
+            arrow.Arrow.interval(
+                "hour",
+                datetime(2013, 5, 5, 12, 30),
+                datetime(2013, 5, 5, 17, 15),
+                4,
+                exact=True,
+            )
+        )
+
+        expected = [
+            (
+                arrow.Arrow(2013, 5, 5, 12, 30),
+                arrow.Arrow(2013, 5, 5, 16, 29, 59, 999999),
+            ),
+            (
+                arrow.Arrow(2013, 5, 5, 16, 30),
+                arrow.Arrow(2013, 5, 5, 17, 14, 59, 999999),
+            ),
+        ]
```

Comment:
Hey @JunhwanK, sorry it has been a while. We have been swamped with work and outside commitments, but we are looking to get this merged now.

On the current version of Arrow, I get this:

```shell
>>> from datetime import datetime
>>> start = datetime(2013, 5, 5, 12, 30)
>>> end = datetime(2013, 5, 5, 17, 15)
>>> for r in arrow.Arrow.interval('hour', start, end, 2):
...     print(r)
...
(<Arrow [2013-05-05T12:00:00+00:00]>, <Arrow [2013-05-05T13:59:59.999999+00:00]>)
(<Arrow [2013-05-05T14:00:00+00:00]>, <Arrow [2013-05-05T15:59:59.999999+00:00]>)
(<Arrow [2013-05-05T16:00:00+00:00]>, <Arrow [2013-05-05T17:59:59.999999+00:00]>)
>>> for r in arrow.Arrow.interval('hour', start, end, 4):
...     print(r)
...
(<Arrow [2013-05-05T12:00:00+00:00]>, <Arrow [2013-05-05T15:59:59.999999+00:00]>)
>>> arrow.__version__
'0.15.5'
```

But on your branch I get this:

```shell
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import datetime
>>> import arrow
>>> start = datetime(2013, 5, 5, 12, 30)
>>> end = datetime(2013, 5, 5, 17, 15)
>>> for r in arrow.Arrow.interval('hour', start, end, 4, exact=False):
...     print(r)
... 
(<Arrow [2013-05-05T12:00:00+00:00]>, <Arrow [2013-05-05T15:59:59.999999+00:00]>)
(<Arrow [2013-05-05T16:00:00+00:00]>, <Arrow [2013-05-05T17:59:59.999999+00:00]>)
>>> for r in arrow.Arrow.interval('hour', start, end, 4, exact=True):
...     print(r)
... 
(<Arrow [2013-05-05T12:30:00+00:00]>, <Arrow [2013-05-05T16:29:59.999999+00:00]>)
(<Arrow [2013-05-05T16:30:00+00:00]>, <Arrow [2013-05-05T17:14:59.999999+00:00]>)
```

The weird thing is that when you have an interval of 4 and exact is false, you only extend to `17:59:59` rather than the expected value of `19:59:59`.

Any idea where this bug could stem from? There may be another bug at play here.